src/testrender/background.h

```diff
@@ -10,17 +10,49 @@
 
 OSL_NAMESPACE_ENTER
 
+#ifdef __CUDACC__
+// std::upper_bound is not supported in device code, so define a version of it here.
+// Adapted from the LLVM Project, see https://llvm.org/LICENSE.txt for license information.
+template<typename T>
+inline OSL_HOSTDEVICE const T*
+upper_bound_cuda(const T* data, int count, const T value)
+{
+    const T* __first = data;
+    const T __value_ = value;
+    int __len        = count;
+    while (__len != 0) {
+        int __l2     = __len / 2;
+        const T* __m = __first;
+        __m += __l2;  // TODO: This is possibly unsafe, should be something like std::advance
```

Comment:
What is unsafe about this?

Reply:
I'm not sure if `std::advance` provides any bounds checking or other such safeguards, but if that's not a concern I'll drop the comment.